app/src/main/java/com/example/trempel/PdpFragment.kt

```diff
@@ -0,0 +1,111 @@
+package com.example.trempel
+
+import android.Manifest
+import android.app.Activity.RESULT_OK
+import android.app.AlertDialog
+import android.content.pm.PackageManager
+import android.os.Bundle
+import android.os.Environment
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import androidx.activity.result.ActivityResultLauncher
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.core.content.ContextCompat
+import androidx.fragment.app.Fragment
+import java.io.File
+import androidx.core.app.ActivityCompat.startActivityForResult
+
+import android.content.Intent
+import android.media.ImageReader
+import android.net.Uri
+import android.provider.MediaStore
+
+
+class PdpFragment : Fragment() {
+
+    private lateinit var iVProduct: ImageView
+
+    private var requestPermissionLauncher: ActivityResultLauncher<String>? = null
+
+        override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        requestPermissionLauncher =
+            registerForActivityResult(ActivityResultContracts.RequestPermission()) { isGranted: Boolean ->
+                if (isGranted) {
+                    readFile()
+                } else {
+                    createAlertDialog()
+                }
+            }
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        val rootview = inflater.inflate(R.layout.pdp_fragment, container, false)
+        iVProduct = rootview.findViewById(R.id.iV_product)
+        iVProduct.setOnClickListener {
+            hasReadExternalStoragePermission()
+        }
+        return rootview
+    }
+
+        private fun hasReadExternalStoragePermission() {
+        requestPermissionLauncher
```

Comment:
something weird has happened to formatting, please refer https://kotlinlang.org/docs/coding-conventions.html#functions

`You can tap Command + Alt + L to autoformat your code in some cases`